dbms/src/Parsers/ParserInsertQuery.cpp

```diff
@@ -34,26 +37,37 @@ bool ParserInsertQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
     ParserToken s_rparen(TokenType::ClosingRoundBracket);
     ParserIdentifier name_p;
     ParserList columns_p(std::make_unique<ParserCompoundIdentifier>(), std::make_unique<ParserToken>(TokenType::Comma), false);
+    ParserFunction table_function_p;
 
     ASTPtr database;
     ASTPtr table;
     ASTPtr columns;
     ASTPtr format;
     ASTPtr select;
+    ASTPtr table_function;
     /// Insertion data
     const char * data = nullptr;
 
     if (!s_insert_into.ignore(pos, expected))
         return false;
 
-    if (!name_p.parse(pos, table, expected))
-        return false;
-
-    if (s_dot.ignore(pos, expected))
+    if (s_function.ignore(pos, expected) || (s_table.ignore(pos, expected) && s_function.ignore(pos, expected)))
```

Comment:
If we introduced syntax `INSERT INTO FUNCTION f()`, `INSERT INTO TABLE FUNCTION f()`, we must also add support for `INSERT INTO TABLE t` in addition to `INSERT INTO t` for consistency.